Weather-Monitoring-And-Reporting-Service_Tests/PublisherTest/WeatherPublisherTest.cs

```diff
@@ -0,0 +1,122 @@
+﻿using AutoFixture.AutoMoq;
+using AutoFixture;
+using Moq;
+using Weather_Monitoring_And_Reporting_Service;
+using Weather_Monitoring_And_Reporting_Service.Configuration;
+using Weather_Monitoring_And_Reporting_Service.Publisher;
+using Weather_Monitoring_And_Reporting_Service.Strategies;
+using Weather_Monitoring_And_Reporting_Service.Subscriber;
+using Xunit;
+using Weather_Monitoring_And_Reporting_Service.WeatherBot;
+
+namespace Weather_Monitoring_And_Reporting_Service_Tests.PublisherTest;
+
+public class WeatherPublisherTest
+{
+    private readonly IFixture fixture;
+
+    public WeatherPublisherTest()
+    {
+        fixture = new Fixture().Customize(new AutoMoqCustomization());
+    }
+
+    private WeatherPublisher CreateWeatherDataPublisher()
+    {
+        var mockTextFormatStrategy = new Mock<ITextFormatStrategy>();
```

Comment:
Make them as scoped variables, the dependencies, also use fixtures to create the objects.